File: 1/haskell/one.hs
main = do print $ solution 1000 3 5   
solution x y z = multsum (x-1) y + multsum (x-1) z - multsum (x-1) (y*z)
multsum x y
    | x < y          = 0
    | mod x y == 0   = x+(multsum (x-y) y)
    | otherwise      = multsum (x-1) y 

File: 1/java/One.java
public class One
{
	public static void main(String[] args)
	{
		System.out.println(solution(1000, 3, 5));
	}
	public static int solution(int x, int y, int z)
	{
		x--;
		return y*sum(x/y) + z*sum(x/z) - (y*z)*sum(x/(y*z));
	}
	public static int sum(int x)
	{
		if (x==0)
			return 0;
		return x+sum(x-1);
	}
}